src/core_modules/capture-core/components/DataEntries/common/useMetadataForRegistrationForm/index.js

```diff
@@ -0,0 +1,2 @@
+// @flow
+export { useMetadataForRegistrationForm } from './useMetadataForRegistrationForm';
```

Comment:
Since `useMetadataForRegistrationForm` is used in the DataEntries and Pages components, I propose to move it to a common place in the file hierarchy. For example in the [capture-core/hooks](https://github.com/dhis2/capture-app/tree/master/src/core_modules/capture-core/hooks) folder. WDYT?

Reply:
I might be missing some nuances here, but I think `TEIregistrationEntry` and `EnrollmentRegistrationEntry` should be responsible for getting the data that is retrieved from `useMetadataForRegistrationForm`. At the moment we are also invoking  `useMetadataForRegistrationForm` from some of the Pages-components (like you say), but at first glance it seems like it adds quite a lot of unnecessary complexity and confusion? I can see we have to do some minor refactoring, but would be great to clean this up. And at that point I would put the files here: `capture-core/components/DataEntries/common/TEIAndEnrollment`.

Great if you can have a look @eirikhaugstulen.

Reply:
@JoakimSM Not sure if I understood everything you meant here, but tried to move all usages of `useMetadataForRegistrationPage` inside `TEIregistrationEntry` and `EnrollmentRegistrationEntry`. Had to comment out a flow error in `NewRelationship` as it was not part of the current rewrite. Let me know if it was not what you intended 👍